.ci/run-opensearch.sh

```diff
@@ -58,40 +58,62 @@ END
   local_detach="true"
   if [[ "$i" == "$((NUMBER_OF_NODES-1))" ]]; then local_detach=$DETACH; fi
 
-  echo -e "\033[34;1mINFO: building $CLUSTER container\033[0m"
-  docker build \
-    --file=.ci/$CLUSTER/Dockerfile \
-    --build-arg SECURE_INTEGRATION=$SECURE_INTEGRATION \
-    --build-arg OPENSEARCH_VERSION=$OPENSEARCH_VERSION \
-    --tag=$CLUSTER-secure-$SECURE_INTEGRATION \
-    .
-
-  echo -e "\033[34;1mINFO:\033[0m Starting container $node_name \033[0m"
   set -x
   healthcmd="curl -vvv -s --fail http://localhost:9200/_cluster/health || exit 1"
   if [[ "$SECURE_INTEGRATION" == "true" ]]; then
     healthcmd="curl -vvv -s --insecure -u admin:admin --fail https://localhost:9200/_cluster/health || exit 1"
   fi
 
-  docker run \
-    --name "$node_name" \
-    --network "$network_name" \
-    --env "ES_JAVA_OPTS=-Xms1g -Xmx1g" \
-    "${environment[@]}" \
-    "${volumes[@]}" \
-    "${security[@]}" \
-    --publish "$http_port":9200 \
-    --ulimit nofile=65536:65536 \
-    --ulimit memlock=-1:-1 \
-    --detach="$local_detach" \
-    --health-cmd="$(echo $healthcmd)" \
-    --health-interval=2s \
-    --health-retries=20 \
-    --health-timeout=2s \
-    --rm \
-    -d \
-    $CLUSTER-secure-$SECURE_INTEGRATION;
+  if [[ "$IS_UNRELEASED" == "false" ]]; then
+    echo -e "\033[34;1mINFO: building $CLUSTER container\033[0m"
+    docker build \
+      --file=.ci/$CLUSTER/Dockerfile \
+      --build-arg SECURE_INTEGRATION=$SECURE_INTEGRATION \
+      --build-arg OPENSEARCH_VERSION=$OPENSEARCH_VERSION \
+      --tag=$CLUSTER-secure-$SECURE_INTEGRATION \
+      .
+
+    echo -e "\033[34;1mINFO:\033[0m Starting container $node_name \033[0m"
 
+    docker run \
+      --name "$node_name" \
+      --network "$network_name" \
+      --env "ES_JAVA_OPTS=-Xms1g -Xmx1g" \
+      "${environment[@]}" \
+      "${volumes[@]}" \
+      "${security[@]}" \
+      --publish "$http_port":9200 \
+      --ulimit nofile=65536:65536 \
+      --ulimit memlock=-1:-1 \
+      --detach="$local_detach" \
+      --health-cmd="$(echo $healthcmd)" \
+      --health-interval=2s \
+      --health-retries=20 \
+      --health-timeout=2s \
+      --rm \
+      -d \
+      $CLUSTER-secure-$SECURE_INTEGRATION;
+  else
+    docker run \
+      --name "$node_name" \
+      --network "$network_name" \
+      --env "ES_JAVA_OPTS=-Xms1g -Xmx1g" \
+      "${environment[@]}" \
+      "${volumes[@]}" \
+      "${security[@]}" \
+      --publish "$http_port":9200 \
+      --ulimit nofile=65536:65536 \
+      --ulimit memlock=-1:-1 \
+      --detach="$local_detach" \
+      --health-cmd="$(echo $healthcmd)" \
+      --health-interval=2s \
+      --health-retries=20 \
+      --health-timeout=2s \
+      --rm \
+      -d \
+      opensearch:test;
+  fi
```

Comment:
```suggestion
CLUSTER_TAG = $CLUSTER
  if [[ "$IS_UNRELEASED" == "false" ]]; then
    CLUSTER_TAG = $CLUSTER_TAG-secure-$SECURE_INTEGRATION
    echo -e "\033[34;1mINFO: building $CLUSTER container\033[0m"
    docker build \
      --file=.ci/$CLUSTER/Dockerfile \
      --build-arg SECURE_INTEGRATION=$SECURE_INTEGRATION \
      --build-arg OPENSEARCH_VERSION=$OPENSEARCH_VERSION \
      --tag=$CLUSTER_TAG \
      .
 else
   CLUSTER_TAG = $CLUSTER_TAG:test
 fi
 echo -e "\033[34;1mINFO:\033[0m Starting container $node_name \033[0m"

  docker run \
    --name "$node_name" \
    --network "$network_name" \
    --env "ES_JAVA_OPTS=-Xms1g -Xmx1g" \
    "${environment[@]}" \
    "${volumes[@]}" \
    "${security[@]}" \
    --publish "$http_port":9200 \
    --ulimit nofile=65536:65536 \
    --ulimit memlock=-1:-1 \
    --detach="$local_detach" \
    --health-cmd="$(echo $healthcmd)" \
    --health-interval=2s \
    --health-retries=20 \
    --health-timeout=2s \
    --rm \
    -d \
    $CLUSTER_TAG;
```

Reply:
For unreleased OpenSearch, the docker image gets built when the `./gradlew assemble` runs for OpenSearch, it is built with tag `opensearch:test`. So when unreleased is true, we only need to do the docker run, docker build happens in the CI workflow. When unreleased is false, we get build the docker image using Dockerfile and then run it.

Reply:
Ah I misunderstood the refactor, this makes sense! I will make the change.